pylint/checkers/base.py

```diff
@@ -1159,12 +1159,14 @@ def _check_using_constant_test(self, node, test):
 
     def visit_module(self, _: nodes.Module) -> None:
         """check module name, docstring and required arguments"""
+        assert isinstance(self.stats["module"], int)
```

Comment:
In tests we can handle it this way, but we  can't do this in pylint's code. We probably need to remove the possibility that it's not an int altogether by doing a refactor. Crashing with an assert will still be a crash.

Reply:
Are you sure? Based on the code on L1163 I think `self.stats["module"]` will always be an `int`. Otherwise it would already produce crashes (and none seem to be reported).
I think this actually asserts something that is already always the case but is needed for `mypy` not to fail (and as a helpful remainder for us that this stat should always be an `int`).

Reply:
It makes mypy stop warning about the issue but it's the wrong way to fix the issue imo.
- It lower performance because we'll now make an assert at run time
- The root cause of ``self.stats`` not being always properly initialized and ``visit_module`` not being safely callable in badly initialized child class is not addressed (default value for ``self.stats`` is None).
- The error is only marginally better as we'd have a ``TypeError: can only concatenate str (not "int") to str`` already or something similarly clear.
- This is not the case here, but in general for pylint this kind of issue could come from not anticipating a strange code construct that will come from astroid. For example we think we can have only an inner function, but it happens the user defined an inner class. Asserting that we have a ``FunctionDef`` when we have a ``ClassDef`` instead will still crash pylint the same way that if we did nothing at all. It's just silencing the linter with an assumption we made.

I think this is a design issue on the base checker that would require a refactor and we may as well just ignore it until we do the refactor.